darts/tests/models/forecasting/test_local_forecasting_models.py

```diff
@@ -336,3 +338,43 @@ def test_statsmodels_dual_models(self):
             # check backtesting with retrain=False
             model: TransferableDualCovariatesForecastingModel = model_cls(**kwargs)
             model.backtest(series1, future_covariates=exog1, retrain=False)
+
+    def test_backtest_retrain(self):
```

Comment:
You could also add a test to make sure `retrain` is being called as expected (with expected arguments, e.g., past/future covariates etc) when it is a `Callable`. I think this should be pretty easy to do with `unittest.mock`, you can check out examples here: https://docs.python.org/3/library/unittest.mock.html

Reply:
I wasn't able to use `Mock`, yet implemented a test with "fake" `train_series`, `past_covariates` and `future_covariates` all equal to the series itself, but testing on some "proper" condition.

Reply:
This can also work, however your current test does not test correctness, or even that the function has been called. If you want to do it this way (using conditions on the series within the function), I would maybe cook a small toy example where the result of the backtest is not the same depending (for instance) on the quantile quantities computed on the covariates series. This way you could then test that 
* The output of the backtesting procedure is not the same if different `future_covariates` are used
* The output is as you expect in some pre-computed case (e.g., the historical forecasts is a series containing values [x, y, z]).

That being said, I still think it might be easier to simply test whether the function is called by using mocking. You can find an example of how we are doing it in other tests, for instance [here](https://github.com/unit8co/darts/blob/master/darts/tests/models/forecasting/test_torch_forecasting_model.py#L51), where we then check [here](https://github.com/unit8co/darts/blob/master/darts/tests/models/forecasting/test_torch_forecasting_model.py#L82) whether a given function has been called or not.

Reply:
Just to explain myself: it's easy to mock a retrain function (i.e. a callable returning a boolean), but the function itself is called at each iteration (and I am able to check for that). 
I would be more interested to verify that `_fit_wrapper` method is called multiple times. 
However when trying to patch/mock `_fit_wrapper` and `_predict_wrapper`, I get all sort of errors.

Reply:
@hrzn do you have any new suggestion on this?

Reply:
Then if mocking does not work my suggestion would be to at least test that the output of `model_cls.backtest()` is what you expect it to be for a couple of simple cases (e.g., when you retrain each 1st of the month).

Reply:
Found few tricks to make it work using `mock.patch`. Looking forward to the review.